File: 232-implement-queue-using-stacks/implement-queue-using-stacks.cpp
class MyQueue {
public:
    stack<int>st;
    stack<int>helper;

    MyQueue() {
        
    }
    
    void push(int x) {
        st.push(x);
    }
    
    int pop() {
        if(helper.empty())
        {

        while(!st.empty())
        {
            helper.push(st.top());
            st.pop();
        }
            }

        int val = helper.top();
        helper.pop();

        return val;
    }
    
    int peek() {
        if(helper.empty())
        {
            while(!st.empty())
            {
                helper.push(st.top());
                st.pop();
            }
        }

        int val = helper.top();
        return val;
    }
    
    bool empty() {
        return st.empty() && helper.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 232-implement-queue-using-stacks/implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.st = []
        self.helper = []
        

    def push(self, x: int) -> None:
        self.st.append(x)

    def pop(self) -> int:
        self.peek()
        return self.helper.pop()
        

    def peek(self) -> int:
        if not self.helper:
            while self.st:
                self.helper.append(self.st.pop())

        return self.helper[-1]
        

    def empty(self) -> bool:
        return not self.st and not self.helper
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()